ledger/src/blockstore_db.rs

```diff
@@ -484,7 +499,33 @@ impl Rocks {
             new_cf_descriptor::<ProgramCosts>(options, oldest_slot),
             new_cf_descriptor::<OptimisticSlots>(options, oldest_slot),
             new_cf_descriptor::<MerkleRootMeta>(options, oldest_slot),
-        ]
+        ];
+
+        // The default column is handled automatically, we don't need to create
+        // a descriptor for it
+        const DEFAULT_COLUMN_NAME: &str = "default";
+        let known_cfs: HashSet<_> = cf_descriptors
+            .iter()
+            .map(|cf_descriptor| cf_descriptor.name().to_string())
+            .chain(std::iter::once(DEFAULT_COLUMN_NAME.to_string()))
+            .collect();
+        detected_cfs.iter().for_each(|cf_name| {
+            if known_cfs.get(cf_name.as_str()).is_none() {
+                info!("Detected unknown column {cf_name}, opening column with basic options");
+                // This version of the software was unaware of the column, so
+                // it fair to assume that we will not attempt to read or write
+                // the column. So, set some bare bones settings to avoid
+                // using extra resources on this unknown column.
+                let mut options = Options::default();
+                // Lower the default to avoid unnecessary allocations
+                options.set_write_buffer_size(1024 * 1024);
+                // Disable compactions to avoid any modifications to the column
+                options.set_disable_auto_compactions(true);
+                cf_descriptors.push(ColumnFamilyDescriptor::new(cf_name, options));
```

Comment:
Technically, we only need this when `AccessType` is `Primary` or `PrimaryForMaintenance`.

So, I could put the logic to read the existing descriptors and create the missing ones within a match statement that checks if we're opening with `Primary`; thoughts ?

Reply:
Works for me